### ISSUE TITLE
Snapshot has incorrect corner mask when using `layer.maskedCorners`

### ISSUE THREAD

**Issue by ryanmeisters** [pointfreeco/swift-snapshot-testing#358]
Thanks for an awesome library! I've been enjoying it immensely since adding to my projects last week 🙌 

I'm not sure if this is a UIKit issue or a SnapshotTesting issue, but I get an incorrect snapshot when using `CALayer`'s `maskedCorners` property (introduced in iOS 11)

e.g. 
```
func test_maskedCorner() {
    let v = UIView()
    v.frame = CGRect(x: 0, y: 0, width: 100, height: 40)
    v.backgroundColor = .blue
    v.layer.cornerRadius = 12
    v.layer.maskedCorners = [.layerMaxXMaxYCorner, .layerMinXMaxYCorner]
    assertSnapshot(matching: v, as: .image, record: true)
  }
```

This should produce a blue rectangle with the bottom two corners rounded, but instead produces the image as if `maskedCorners` was not applied.

Result: 
![test_sdfads 1](https://user-images.githubusercontent.com/1131791/85055908-35215400-b153-11ea-891d-6fcd17a8289b.png)

**Comment by chefnobody** [comment]
Shootin' from the hip a little bit here, but I wonder if this has anything to do w/ the `snapshotView` function that attempts to render the view's layer into the provided graphics context and a warning in the `CALayer` docs for that function call.

```
view.layer.render(in: ctx.cgContext)
```
(line 716 of View.swift)

If you explore the CALayer `render` method docs they say this:

```
    /* Renders the receiver and its sublayers into 'ctx'. This method
     * renders directly from the layer tree. Renders in the coordinate space
     * of the layer.
     *
     * WARNING: currently this method does not implement the full
     * CoreAnimation composition model, use with caution. */
```
Maybe this warning has something to do with it? 

Alternatively, I haven't yet tried the `drawHeirarchy(in:afterScreenUpdates:)` approach that draws the view into a key window, as my project uses a framework and so it has no host application.

**Comment by chefnobody** [comment]
Going a little further, [I found docs for `render(in:)` online](https://developer.apple.com/documentation/quartzcore/calayer/1410909-render) whose warning is more explicit:

> Important
The OS X v10.5 implementation of this method does not support the entire Core Animation composition model. QCCompositionLayer, CAOpenGLLayer, and QTMovieLayer layers are not rendered. Additionally, layers that use 3D transforms are not rendered, nor are layers that specify backgroundFilters, filters, compositingFilter, or a **_mask values_**. Future versions of macOS may add support for rendering these layers and properties.

Emphasis added ^^

**Comment by jibeex** [comment]
Is there any solution/workaround?

**Comment by Nikoloutsos** [comment]
Facing the same problem. Is there any workaround?

**Comment by dmhts** [comment]
Same here

**Comment by penelopearaujo** [comment]
Did anyone find any solution or workaround? I'm facing the same issue.

**Comment by dk53** [comment]
Won't add a lot of value here, just saying I face the same issue overthere

**Comment by stephencelis** [comment]
As someone brought up previously in this issue, I believe you may need to introduce a host application and set `drawHierarchyInKeyWindow` for this to render properly. Anyone give that a shot?

**Comment by chefnobody** [comment]
@stephencelis I took a crack at this and I think I've verified that using `.image(drawHeirarchyInKeyWindow: true)` does seem to fix the issue.

I think another possible alternative is to try a `UIBezierPath's` special init and then set that as the mask for your view's layer as that seems to work w/o the need for a host app.

At any rate, here's a sample project with a Unit Test target whose host application is the main app target: https://github.com/chefnobody/snapshot-masked-corners

And a screenshot of the results: 
![Screen Shot 2021-08-18 at 11 39 23 AM](https://user-images.githubusercontent.com/303658/129954213-3aa354de-2e74-482a-b124-f4534f04587c.png)

**Comment by domagojstankovic** [comment]
@chefnobody `.image(drawHierarchyInKeyWindow: true)` works for me. Thanks!

**Comment by domagojstankovic** [comment]
@Nikoloutsos you are right, it didn't work when testing on CI so we had to remove those tests. Thank you for mentioning it anyway!

**Comment by alimkhan6** [comment]
@chefnobody thanks for suggest of drawHierarchy method
